clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/gitRepo/GitRepoArtifactCredentials.java

```diff
@@ -44,11 +58,41 @@
   @Getter private final String name;
   private final String username;
   private final String password;
+  private final String token;
+  private final String sshPrivateKeyFilePath;
+  private final String sshPrivateKeyPassphrase;
+  private final String sshKnownHostsFilePath;
+  private final boolean sshTrustUnknownHosts;
+  private final AuthType authType;
+
+  private enum AuthType {
+    HTTP,
+    TOKEN,
+    SSH,
+    NONE
+  }
 
   public GitRepoArtifactCredentials(GitRepoArtifactAccount account) {
     this.name = account.getName();
     this.username = account.getUsername();
     this.password = account.getPassword();
+    this.token = account.getToken();
+    this.sshPrivateKeyFilePath = account.getSshPrivateKeyFilePath();
+    this.sshPrivateKeyPassphrase = account.getSshPrivateKeyPassphrase();
+    this.sshKnownHostsFilePath = account.getSshKnownHostsFilePath();
+    this.sshTrustUnknownHosts = account.isSshTrustUnknownHosts();
+
+    if (!StringUtils.isEmpty(username) && !StringUtils.isEmpty(password)) {
+      authType = AuthType.HTTP;
+    } else if (!StringUtils.isEmpty(token)) {
+      authType = AuthType.TOKEN;
+    } else if (!StringUtils.isEmpty(sshPrivateKeyFilePath)
```

Comment:
Possibly? I copied all of this logic from the App Engine logic so I assume most cases have been accounted for since it's been around longer. If cases like this come up I don't see why we couldn't just relax this condition in a patch.